sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -1919,9 +1920,14 @@ case class ArrayPosition(left: Expression, right: Expression)
        b
   """,
   since = "2.4.0")
-case class ElementAt(left: Expression, right: Expression)
+case class ElementAt(
+    left: Expression,
+    right: Expression,
+    failOnError: Boolean = SQLConf.get.ansiEnabled)
```

Comment:
Why just have `val failOnError: Boolean = SQLConf.get.ansiEnabled`? Do you need to assign it a value other than `SQLConf.get.ansiEnabled` when constructing?

Reply:
Making it a parameter is more robust to retain this info. Otherwise, we may change it when transform and copy the expresion.

This also helps if we want to support SAFE prefix like https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators

Reply:
When doing case class copy, it's better to just copy it from case class, but not re-get from SQLConf.get, it might change. I saw lot of same sample code inside spark.

Reply:
for example